internal/pkg/describe/backend_service.go

```diff
@@ -79,67 +79,118 @@ func (d *BackendServiceDescriber) URI(envName string) (string, error) {
 	return s.String(), nil
 }
 
+type backendSvcStackInfo struct {
+	config           *ServiceConfig
+	serviceDiscovery serviceDiscovery
+	envVars          map[string]string
+	err              error
+}
+
+func (d *BackendServiceDescriber) describeStackInfo(env string) backendSvcStackInfo {
+	svcParams, err := d.svcDescriber[env].Params()
+	if err != nil {
+		return backendSvcStackInfo{err: fmt.Errorf("retrieve service deployment configuration: %w", err)}
+	}
+	config := &ServiceConfig{
+		Environment: env,
+		Port:        svcParams[stack.LBWebServiceContainerPortParamKey],
+		Tasks:       svcParams[stack.ServiceTaskCountParamKey],
+		CPU:         svcParams[stack.ServiceTaskCPUParamKey],
+		Memory:      svcParams[stack.ServiceTaskMemoryParamKey],
+	}
+	sd := serviceDiscovery{
+		Service: d.svc,
+		Port:    svcParams[stack.LBWebServiceContainerPortParamKey],
+		App:     d.app,
+		Env:     env,
+	}
+	backendSvcEnvVars, err := d.svcDescriber[env].EnvVars()
+	if err != nil {
+		return backendSvcStackInfo{err: fmt.Errorf("retrieve environment variables: %w", err)}
+	}
+	return backendSvcStackInfo{
+		config:           config,
+		envVars:          backendSvcEnvVars,
+		serviceDiscovery: sd,
+	}
+}
+
+func (d *BackendServiceDescriber) svcStackResources(envs []string) (map[string][]*CfnResource, error) {
+	resources := make(map[string][]*CfnResource)
+	if d.enableResources {
+		resourceCh := make(chan svcStackResources, len(envs))
+		defer close(resourceCh)
+		for _, env := range envs {
+			go func(env string) {
+				err := d.initServiceDescriber(env)
```

Comment:
hmm tbh I'm a little bit worried about these changes. 
`initServiceDescriber` writes to a `map` in parallel but it doesn't lock while reading and writing to it. We can use [sync.Map](https://golang.org/pkg/sync/#Map) or we should create just brand new objects for each env which is simpler.

Reply:
yeah sync.Map might not be a good idea since it returns interface{} which is annoying because we need to assume type so many times. We might need to create a brand new objects but i was about to do the caching. creating new objects means potentially we lose some performance.